### PR TITLE
Set up gRPC service

### PR DESCRIPTION
**Pull request by hojoung97**
#### Summary
Add gRPC service - implements a RPC called `GetLastSnapshot` that returns latest log info that has been verified by the monitor mirror job
- `internal/monitorGRPC` contains
   - proto files
   - generated gRPC code
   - gRPC server implementation (`monitorGRPCServer.go`)
-  `cmd/grpcServer` contains
   - gRPC server main executable

Other Changes:
- Move log info file reading from `cmd/mirroring/main.go` to `mirroring/utils.go`
- Add consistency check fail error to identify it from other errors (`mirroring/mirroring.go`)

#### Release Note

```release-note

```

### CODE REVIEW

File: cmd/grpcServer/main.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		port = val
+	}
+
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%s", port))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	grpcServer := grpc.NewServer()
+	monitorGRPC.RegisterMonitorServiceServer(grpcServer, monitorGRPC.NewServer())
```
Comment (A0su) [inline cmd/grpcServer/main.go:40]: Will we not need access to the newly instantiated `monitorGRPC.NewServer()` at some point in the future for the argument `srv MonitorServiceServer`?

File: cmd/grpcServer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# limitations under the License.
+
+# Apply multi-stage build
+FROM golang:alpine AS builder
+WORKDIR /builder
+
+# Copy go source files for build
+COPY go.mod ./
+COPY go.sum ./
+COPY cmd/grpcServer/main.go ./
```
Comment (A0su) [inline cmd/grpcServer/Dockerfile:23]: Could coalesce three copies into one but I don't believe there's any benefit at the current moment

File: mirroring/utils.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	// Read line by line and get the last line
+	scanner := bufio.NewScanner(file)
+	line := ""
+	for scanner.Scan() {
+		line = scanner.Text()
+	}
+
+	// Each line is in the format of space-separeted info: "treeSize rootHash"
+	parsed := strings.Split(line, " ")
```
Comment (A0su) [inline mirroring/utils.go:43]: Do we want error handling if `Split()` returns an empty slice?

**Review by A0su — COMMENTED** [review]
Few minor q's but rest looks great and build works properly on my machine

File: mirroring/utils.go (hunk 3 above)
Comment (hojoung97) [reply mirroring/utils.go:43]: Yes please!

File: cmd/grpcServer/Dockerfile (hunk 2 above)
Comment (hojoung97) [reply cmd/grpcServer/Dockerfile:23]: Can you collapse into one? I would like to see how that's done :)

File: cmd/grpcServer/main.go (hunk 1 above)
Comment (hojoung97) [reply cmd/grpcServer/main.go:40]: Yes, that would be better

File: cmd/grpcServer/Dockerfile (hunk 2 above)
Comment (A0su) [reply cmd/grpcServer/Dockerfile:23]: Yup! The way I usually have written it before is as `COPY go.mod go.sum cmd/grpcServer/main.go ./`.

**Comment by haydentherapper** [comment]
Tracking in https://github.com/sigstore/rekor-monitor/issues/91

**Pull request closed without merging**